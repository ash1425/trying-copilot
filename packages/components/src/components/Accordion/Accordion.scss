.accordion-title {
  background-color: #9daeb1;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 1rem;
}

.accordion-content {
  background-color: #c3d1d4;
  padding: 1rem;
}
