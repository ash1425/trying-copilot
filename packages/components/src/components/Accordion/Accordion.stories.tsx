import React from "react";

import { Meta, Story } from "@storybook/react";

import Accordion, { IAccordionProps } from "./Accordion";

export default {
  component: Accordion,
  title: "components/Accordion",
  argTypes: {
    className: {
      options: "accordion",
      control: { type: "text" },
    },
    title: { options: "section-1", control: { type: "text" } },
    content: {
      options: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
      control: { type: "text" },
    },
  },
} as Meta;

const Template: Story<IAccordionProps> = (args) => {
  const { title, className, content } = args;
  return <Accordion className={className} title={title} content={content} />;
};
export const Default = Template.bind({});

Default.args = {
  className: "accordion",
  title: "section-1",
  content: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
};
