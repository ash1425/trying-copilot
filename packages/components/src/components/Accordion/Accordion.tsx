import React, { useState } from "react";
import "./Accordion.scss";

export interface IAccordionProps {
  className: string;
  title: string;
  content: string;
}

const Accordion = ({ className, title, content }: IAccordionProps) => {
  const [isActive, setIsActive] = useState(false);

  return (
    <div className={`${className}`}>
      <div className={`${className}-item`}>
        <div
          className={`${className}-title`}
          onClick={() => setIsActive(!isActive)}
        >
          <div>{title}</div>
          <div>{isActive ? "-" : "+"}</div>
        </div>
        {isActive && <div className={`${className}-content`}>{content}</div>}
      </div>
    </div>
  );
};

export default Accordion;
