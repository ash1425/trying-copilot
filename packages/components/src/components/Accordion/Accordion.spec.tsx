import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Accordion from "./Accordion";

describe("Accordion", () => {
  test("renders Accordion component", () => {
    render(
      <Accordion
        className="Accordion"
        title="section 1"
        content="Lorem ipsum dolor sit amet, consectetur adipiscing elit"
      />
    );
  });

  test("should have the title", () => {
    render(
      <Accordion
        className="Accordion"
        title="section 1"
        content="Lorem ipsum dolor sit amet, consectetur adipiscing elit"
      />
    );
    expect(screen.getByText("section 1")).toHaveTextContent("section 1");
  });

  test("should have the content", () => {
    const { asFragment } = render(
      <Accordion
        className="Accordion"
        title="section 1"
        content="Lorem ipsum dolor sit amet, consectetur adipiscing elit"
      />
    );
    expect(screen.getByText("section 1")).toHaveTextContent("section 1");
    userEvent.click(screen.getByText("section 1"));
    expect(
      screen.getByText(
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit"
      )
    ).toHaveTextContent(
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit"
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
