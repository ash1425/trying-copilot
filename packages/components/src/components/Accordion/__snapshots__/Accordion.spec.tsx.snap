// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should have the content 1`] = `
<DocumentFragment>
  <div
    class="Accordion"
  >
    <div
      class="Accordion-item"
    >
      <div
        class="Accordion-title"
      >
        <div>
          section 1
        </div>
        <div>
          -
        </div>
      </div>
      <div
        class="Accordion-content"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit
      </div>
    </div>
  </div>
</DocumentFragment>
`;
